src/tasks/task.ts

```diff
@@ -0,0 +1,86 @@
+import assign from "../util/assign";
+import { generateId } from "../util/unique-id";
+import { beacon } from "../util/beacon";
+import { getClientInfo } from "../util/client-info";
+
+interface State {
+  hasRan: boolean;
+}
+
+class Task {
+  public config: any = {};
+  public result: any = {};
+  public state: State = { hasRan: false };
+
+  public constructor(config: any) {
+    // TODO: more strict typing here
+    this.config = assign({}, config, { testId: generateId() });
```

Comment:
I'd be interested in seeing what preset-env does here if we use `Object.assign` and drop the `assign()` import.

Reply:
will look into this.

Reply:
It's not that clear to me:
Using `Object.assign()`:
```
-rw-r--r--  1 psbanka  staff  24195 Jun  5 13:24 dist/90dd1549dc2c2e4a0906.worker.js
-rw-r--r--  1 psbanka  staff  12824 Jun  5 13:24 dist/main.037da71c28d93d36160a.js
```
using our own polyfill:
```
-rw-r--r--  1 psbanka  staff  23814 Jun  5 13:23 dist/d9e41bb3a50a631b8c8f.worker.js
-rw-r--r--  1 psbanka  staff  12824 Jun  5 13:23 dist/main.da10accfecb4d70dc2d2.js
```
So we're saving 381 blocks using your polyfill.